### PR TITLE
Add square spiral plan

### PR DESCRIPTION
**Pull request by awalter-bnl**

### CODE REVIEW

File: bluesky/plan_patterns.py [hunk 1]
```diff
@@ -62,6 +62,120 @@ def spiral(x_motor, y_motor, x_start, y_start, x_range, y_range, dr, nth, *,
...
+        num_st=1
+        if y_num%2==0:
+            y_num+=1
+
+        offset=0
+        
+        x_points.append(x_centre)
+        y_points.append(y_centre)
+        
+    #print( 'x_range = '+str(x_range)+', x_num = '+str(x_num)  )
```
Comment (danielballan) [inline bluesky/plan_patterns.py]: It's a minor thing, but please delete the comment print statements and the blocks of blank lines. (For readability, use occasional blank lines to taste but avoid blocks like the three blank lines below.)

File: bluesky/plan_patterns.py [hunk 2]
```diff
@@ -62,6 +62,120 @@ def spiral(x_motor, y_motor, x_start, y_start, x_range, y_range, dr, nth, *,
     cyc += cycler(y_motor, y_points)
     return cyc
 
+        
+def spiral_square_pattern(x_motor, y_motor, x_centre, y_centre, x_range, y_range, x_num,y_num):
+    '''Square spiral scan, centered around (x_start, y_start)
+    Parameters
```
Comment (danielballan) [inline bluesky/plan_patterns.py]: A blank line is needed above this heading to make the Sphinx documentation render correctly. Likewise for the other sections below.

**Comment by awalter-bnl** [comment]
I have made the requested changes, please look over again.

File: bluesky/plan_patterns.py [hunk 3]
```diff
@@ -62,6 +62,102 @@ def spiral(x_motor, y_motor, x_start, y_start, x_range, y_range, dr, nth, *,
...
+    delta_x = x_range/(x_num-1)
+    delta_y = y_range/(y_num-1)
+    num_ring = max(x_num,y_num)
+    x_max=x_centre + delta_x * (x_num-1)/2
+    x_min=x_centre - delta_x * (x_num-1)/2
+    y_max=y_centre + delta_y * (y_num-1)/2
+    y_min=y_centre - delta_y * (y_num-1)/2
+    
+    for n,i_ring in enumerate(range(num_st, num_ring+1,2)):
+        #print ('x_centre: '+str(x_centre)+', delta_x: '+str(delta_x))
```
Comment (danielballan) [inline bluesky/plan_patterns.py]: Some stray prints are here.

File: bluesky/plan_patterns.py [hunk 4]
```diff
@@ -62,6 +62,102 @@ def spiral(x_motor, y_motor, x_start, y_start, x_range, y_range, dr, nth, *,
...
+    x_range : float
+        x width of spiral
+    y_range : float
+        y width of spiral
+    x_num : float
+        number of x axis points
+    y_num : float (must be even if x_num is even and must be odd if x_num is odd, if not it is increased by 1 to ensure this)
+        number of y axis points
+   
+   Returns
```
Comment (danielballan) [inline bluesky/plan_patterns.py]: This is not aligned (missing just one space) which will break the sphinx rendering. It's fussy!

File: bluesky/plans.py [hunk 5]
```diff
@@ -2887,7 +2887,65 @@ def relative_spiral(detectors, x_motor, y_motor, x_range, y_range, dr, nth,
                               x_range, y_range, dr, nth, tilt=tilt,
                               per_step=per_step, md=_md))
 
+def spiral_square(detectors, x_motor, y_motor, x_centre, y_centre, x_range,
+                  y_range, x_num, y_num, *, per_step=None, md=None):
+    '''Absolute square spiral scan, centered around (x_centre, y_centre)
+    Parameters
```
Comment (danielballan) [inline bluesky/plans.py]: The sections headings in this function also need blank line above them.

**Comment by awalter-bnl** [comment]
OK so clearly I am not as thorough as I thought. I think I have found everything now, please check again.

**Comment by danielballan** [comment]
Looks good. I think the plan code is ready.

The next step, before merging, would normally be adding automated tests. They will be run on all future pull requests to ensure that future, well-intended changes by others who don't understand your code as well don't accidentally break its expected behavior. Your choices are

1. Declare victory at this point and pass the baton to me. (I'd be happy to add the test.)
2. Ask me to provide you with more guidance about how to add a test yourself.

Whichever you like! No one would fault you for punting here in your first bluesky PR. :- )

**Comment by danielballan** [comment]
Superceded by #675

**Comment by danielballan** [comment]
@awalter-bnl This PR is from your master branch into the upstream master branch. Normally we work from a different branch (so-called "feature branch"). One advantage of this workflow is that others =like me can push commits onto the branch to help, whereas it would be impolite of me to push to your master branch -- and probably not allowed by GitHub. I'm going to open a new PR with your commits plus some commits from me adding a test.

**Pull request closed without merging**